### PR TITLE
[PeerList][Part 4] Refactor http.Outbound

### PR DESCRIPTION
**Pull request by willhug**
Summary: In order to put the Peer Logic into the HTTP.outbound as
seamlessly as possible I ended up refactoring some of the outbound
"call" code to be more modular, there should be no functional change in
this PR.

### CODE REVIEW

File: transport/http/outbound.go [hunk 1]
```diff
@@ -121,35 +180,16 @@ func (o *outbound) createSpan(ctx context.Context, req *http.Request, treq *tran
...
-	req.Header = applicationHeaders.ToHTTPHeaders(treq.Headers, nil)
-
 	tracer.Inject(
 		span.Context(),
 		opentracing.HTTPHeaders,
 		opentracing.HTTPHeadersCarrier(req.Header),
 	)
 
-	return ctx, span
+	return ctx, req, span
```
Comment (abhinav) [inline transport/http/outbound.go]: not sure why this is returning the request? it's modifying it in-place.

File: transport/http/outbound.go [hunk 2]
```diff
@@ -169,56 +209,34 @@ func (o *outbound) Call(ctx context.Context, treq *transport.Request) (*transpor
...
-		span.LogEvent(err.Error())
-		if err == context.DeadlineExceeded {
-			return nil, errors.ClientTimeoutError(treq.Service, treq.Procedure, deadline.Sub(start))
-		}
-
-		return nil, err
-	}
-
-	span.SetTag("http.status_code", response.StatusCode)
+	return req
```
Comment (abhinav) [inline transport/http/outbound.go]: Same here. req is modified in-place. Doesn't need to be returned.

File: transport/http/outbound.go (hunk 1 above)
Comment (willhug) [inline transport/http/outbound.go]: this is a bit of a personal styling thing, pass by references mutations can obfuscate that mutations happen in a function.  I realize returning the req doesn't reaaaaaally do anything, it was more for callsite explicitness.  I can remove them if we want.

Comment (abhinav) [inline transport/http/outbound.go]: I don't feel strongly about it. You can leave it as-is.

**Pull request merged**